client/src/services/cryptpad.ts

```diff
@@ -0,0 +1,134 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { Env } from '@/services/environment';
+
+export enum CryptpadDocumentType {
+  Pad = 'pad',
+  Sheet = 'sheet',
+  Doc = 'doc',
+  Presentation = 'presentation',
+  Code = 'code',
+  Unsupported = 'unsupported',
+}
+
+export const ENABLED_DOCUMENT_TYPES = [
+  CryptpadDocumentType.Pad,
+  CryptpadDocumentType.Sheet,
+  CryptpadDocumentType.Doc,
+  CryptpadDocumentType.Presentation,
+  CryptpadDocumentType.Code,
+];
+
+interface CryptpadConfig {
+  document: {
+    url: string;
+    fileType: string;
+    title: string;
+    key?: string;
+  };
+  documentType: CryptpadDocumentType;
+  editorConfig: {
+    lang: string;
+  };
+  autosave: number;
+  events: {
+    onSave: (file: Blob, callback: () => void) => void;
+    onNewKey?: (data: { new: string; old: string }, callback: (key: string) => void) => void;
+    onHasUnsavedChanges?: (unsaved: boolean) => void;
+  };
+}
+
+export class Cryptpad {
+  private scriptId = 'cryptpad-api-js';
+  private containerElement: HTMLElement;
+  private script: HTMLScriptElement;
+  private scriptLoaded = false;
+
+  constructor(containerElement: HTMLElement, serverUrl: string) {
+    this.containerElement = containerElement;
+    if (!Env.isEditicsEnabled()) {
+      throw new Error('Editics is not enabled. Cannot initialize CryptPad.');
+    }
+
+    let script = document.getElementById(this.scriptId) as HTMLScriptElement | null;
+    if (!script) {
+      script = document.createElement('script');
+      if (!script) {
+        throw new Error('Failed to create script element for CryptPad API.');
+      }
+      script.id = this.scriptId;
+      script.src = `${serverUrl}/cryptpad-api.js`;
+      script.async = true;
+      document.head.appendChild(script);
+    }
+    this.script = script;
+  }
+
+  async init(): Promise<void> {
+    if (this.scriptLoaded) return;
+    await new Promise<void>((resolve, reject) => {
+      this.script.onload = (): void => {
+        if (typeof (window as any).CryptPadAPI === 'function') {
```

Comment:
What happens if that's not the case? Maybe just add a little log+call to reject() instead of resolve(), I don't think it'll ever happen but it will be painful to debug.